### PR TITLE
[GPU] Implement deconvolution internal op

### PR DESCRIPTION
**Pull request by dnkurek**
### Details:
 - *Implement deconvolution internal op*
 - *Implement graph transform optimization for deconvolution internal op*
 - *Simplified deconvolution interfaces*
 - *Added more tests for deconvolution*

### CODE REVIEW

File: src/plugins/intel_gpu/include/intel_gpu/primitives/deconvolution.hpp [hunk 1]
```diff
@@ -377,6 +462,10 @@ struct deconvolution : public primitive_base<deconvolution> {
     const primitive_id_arr weights;
     /// @brief List of primitive ids containing bias data.
     const primitive_id_arr bias;
+    /// @brief Tensor with shape of output layout.
+    const primitive_id output_shape;
+    /// @brief The pad type for automatically computing padding sizes
+    ov::op::PadType auto_pad = ov::op::PadType::NOTSET;
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/include/intel_gpu/primitives/deconvolution.hpp:468]: Is it used somewhere?

Comment (dnkurek) [reply src/plugins/intel_gpu/include/intel_gpu/primitives/deconvolution.hpp]: Not really, deleted it

File: src/plugins/intel_gpu/include/intel_gpu/primitives/deconvolution.hpp [hunk 2]
```diff
@@ -377,6 +462,10 @@ struct deconvolution : public primitive_base<deconvolution> {
     const primitive_id_arr weights;
     /// @brief List of primitive ids containing bias data.
     const primitive_id_arr bias;
+    /// @brief Tensor with shape of output layout.
+    const primitive_id output_shape;
```
[the remark this answers is not in the document]
Comment (dnkurek) [reply src/plugins/intel_gpu/include/intel_gpu/primitives/deconvolution.hpp]: Deleted aswell

**Comment by dnkurek** [comment]
build_jenkins

**Comment by dnkurek** [comment]
build_jenkins

File: src/plugins/intel_gpu/src/graph/deconvolution.cpp [hunk 3]
```diff
@@ -82,16 +84,28 @@ layout deconvolution_inst::calc_output_layout(deconvolution_node const& node, ke
                              "As for now, deconvolutions with more than 3 dimensions are not supported");
 
     int32_t x = static_cast<int32_t>(
-        off_factor * pad[pad.size() - 1] + (input_layout.spatial(0) - 1) * strd[strd.size() - 1] + weights_layout.spatial(0));
+        off_factor * pads_begin[pads_begin.size() - 1] +
+        off_factor * pads_end[pads_end.size() - 1] +
+        output_padding[output_padding.size() - 1] +
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/graph/deconvolution.cpp:89]: Why do we need take output_padding into consideration here? I think `x` is supposed to be the size of actual data without any paddings

File: src/plugins/intel_gpu/src/plugin/ops/convolution.cpp [hunk 4]
```diff
@@ -115,7 +121,11 @@ static void CreateConvolutionBackpropDataOp(ProgramBuilder& p, const std::shared
         std::vector<uint16_t> permute_order(weights_rank);
         std::iota(std::begin(permute_order), std::end(permute_order), 0);
         // Should be 1, 0, 2, 3 {, 4} to swap O and I
-        std::swap(permute_order[1], permute_order[0]);
+        if (weights_have_group_dim)
+                std::swap(permute_order[2], permute_order[1]);
+        else
+                std::swap(permute_order[1], permute_order[0]);
```
Comment (sshlyapn) [inline src/plugins/intel_gpu/src/plugin/ops/convolution.cpp:127]: nit:
```suggestion
        if (weights_have_group_dim)
            std::swap(permute_order[2], permute_order[1]);
        else
            std::swap(permute_order[1], permute_order[0]);
```

**Pull request closed without merging**